File: Algorithms/5. Longest Palindromic Substring/python/solution.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        if s == s[::-1]:
            return s

        maxlen = 1
        start = 0
        for i in range(1, len(s)):
            if i - maxlen >= 1:
                temp = s[i - maxlen - 1: i + 1]

                if temp == temp[::-1]:
                    start = i - maxlen - 1
                    maxlen += 2
                    continue
            if i - maxlen >= 0:
                temp = s[i - maxlen: i + 1]

                if temp == temp[::-1]:
                    start = i - maxlen
                    maxlen += 1

        return s[start:start + maxlen]

File: Algorithms/5. Longest Palindromic Substring/swift/solution.swift
class Solution {
    func longestPalindrome(_ s: String) -> String {
        let s = Array(s)
        var result: ArraySlice<Character> = []
        var i = s.startIndex

        while i < s.endIndex - result.count / 2 {
            var left = i
            var right = i
            while right + 1 < s.endIndex && s[right] == s[right + 1] {
                right += 1
            }

            i = right + 1
            while s.startIndex < left && right < s.endIndex - 1 && s[left-1] == s[right+1] {
                left -= 1
                right += 1
            }
            let length = right - left + 1
            if length > result.count {
                result = s[left...right]
            }
        }

        return String(result)
    }
}